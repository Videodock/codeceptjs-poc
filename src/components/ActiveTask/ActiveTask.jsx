import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import Button from '@material-ui/core/Button';
import AppbarTop from '../AppbarTop/AppbarTop';
import AppbarBottom from '../AppbarBottom/AppbarBottom';
import NicknameMenuContainer from '../../containers/NicknameMenuContainer/NicknameMenuContainer';
import styles from './ActiveTask.styles';

const useStyles = makeStyles(styles);

export default function ActiveTask ({ 
  identifier, 
  loading, 
  noTasks, 
  url, 
  campaignIdentifier, 
  onNextTaskButtonClick,
  name, 
  handleNotification,
}) {
  const { t }   = useTranslation('task');
  const classes = useStyles();

  return (
    <React.Fragment>
      <AppbarTop type={name} hasContextNavigation>
        <NicknameMenuContainer campaignIdentifier={campaignIdentifier} />
      </AppbarTop>
      <iframe
        src={url && url.replace('{identifier}', identifier)}
        className={classes.iframe}
        name="task-iframe"
        title={t('task')}
      />
      <AppbarBottom>
        {loading && (
          <Button variant="contained" color="primary" disabled>
            {t('loading')}
          </Button>
        )}
        {noTasks && (
          <Button variant="contained" disabled>
            {t('no_tasks_left')}
          </Button>
        )}
        {!loading && !noTasks && (
          <Button 
            onClick={() => { 
              onNextTaskButtonClick();
              handleNotification('success', t('notifications:thank_you_for_contributing'));
            }} 
            variant="contained" 
            color="primary"
          >
            {t('next_task')}
          </Button>
        )}
      </AppbarBottom>
    </React.Fragment>
  );
}

ActiveTask.propTypes = {
  loading              : PropTypes.bool,
  noTasks              : PropTypes.bool,
  name                 : PropTypes.string,
  url                  : PropTypes.string,
  campaignIdentifier   : PropTypes.string,
  onNextTaskButtonClick: PropTypes.func,
  handleNotification   : PropTypes.func,
};
