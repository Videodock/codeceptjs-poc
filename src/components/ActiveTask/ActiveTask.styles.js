import { createStyles } from '@material-ui/styles';

export default () => createStyles({
  toolbarTop: {
    height        : 56,
    justifyContent: 'space-between',
    color         : '#005AA3',
  },
  appbarBottom: {
    top   : 'auto',
    bottom: 0,
  },
  toolbarBottom: {
    height        : 56,
    justifyContent: 'flex-end',
  },
  logo: {
    width: 138,
  },
  iframe: {
    border: 'none',
    width : '100%',
    height: 'calc(100vh - 112px)',
  },
});
