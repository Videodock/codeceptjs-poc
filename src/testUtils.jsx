import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
