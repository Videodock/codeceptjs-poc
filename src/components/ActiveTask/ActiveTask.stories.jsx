import React from 'react';
import { storiesOf } from '@storybook/react';
import ActiveTask from './ActiveTask';

storiesOf('ActiveTask', module)
  .add('basic', () => <ActiveTask />);
