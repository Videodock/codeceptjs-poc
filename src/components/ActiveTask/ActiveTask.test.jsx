import React from 'react';
import { fireEvent } from '@testing-library/react';
import { render } from '../../testUtils';
import ActiveTask from './ActiveTask';

describe('<ActiveTask />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ActiveTask />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
