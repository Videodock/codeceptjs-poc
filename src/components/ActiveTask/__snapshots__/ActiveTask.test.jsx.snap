// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveTask /> renders and matches snapshot 1`] = `
<header
  class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary"
>
  <div
    class="MuiToolbar-root MuiToolbar-dense makeStyles-toolbarTop-1 MuiToolbar-gutters"
  >
    <img
      alt="logo"
      class="makeStyles-logo-4"
      src="logo.png"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      starticon="[object Object]"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Close
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
